lib/puppet/configurer/plugin_handler.rb

```diff
@@ -36,9 +36,13 @@ def download_plugins(environment)
     locales = Gem::Version.new(server_agent_version) >= SUPPORTED_LOCALES_MOUNT_AGENT_VERSION
     unless locales
       session = Puppet.lookup(:http_session)
-      compiler = session.route_to(:puppet)
-      server_version = session.server_version(compiler.url)
-      locales = server_version && Gem::Version.new(server_version) >= SUPPORTED_LOCALES_MOUNT_AGENT_VERSION
+      compiler = session.service(:puppet)
+      if compiler
+        server_version = session.server_version(compiler.url)
+        if server_version
```

Comment:
This is a bit awkward, might be better to ask the service if it supports a capability, like `compiler.supports?(:locales)` or `compiler.supports_locales?` and rely on `method_missing` for unknown capabilities...